extensions-contrib/azure-extensions/src/main/java/org/apache/druid/storage/azure/AzureStorage.java

```diff
@@ -117,4 +123,26 @@ private CloudBlobContainer getOrCreateCloudBlobContainer(final String containerN
 
     return cloudBlobContainer;
   }
+
+  public ResultSegmentDruid<ListBlobItem> listBlobsWithPrefixInContainerSegmented(
```

Comment:
Do we want a unit test to make sure we're using flat blob listing?

Reply:
Very difficult to do this since the CloudBlobContainer class is final. How do I mock in this case?

Reply:
:( so many final classes in azure - maybe one day we can test with PowerMock 💪